File: Python/the_riddle_practice.py
import time
from selenium import webdriver
browser = webdriver.Chrome()

browser.get('https://techstepacademy.com/trial-of-the-stones')

riddle_of_stone_text_field = browser.find_element_by_id('r1Input')
answer_button_1 = browser.find_element_by_id('r1Btn')
riddle_of_stone_text_field.send_keys('rock')
answer_button_1.click()
answer_1_text = browser.find_element_by_xpath('//*[@id="passwordBanner"]/h4').text

riddle_of_secrets_text_field = browser.find_element_by_id('r2Input')
answer_button_2 = browser.find_element_by_id('r2Butn')
riddle_of_secrets_text_field.send_keys(answer_1_text)
answer_button_2.click()
answer_2_text = browser.find_element_by_xpath('//*[@id="successBanner1"]/h4')

merchant_1_name = browser.find_element_by_xpath('//*[@id="block-05ea3afedc551e378bdc"]/div/div[3]/span/b').text
merchant_1_wealth = browser.find_element_by_xpath('//*[@id="block-05ea3afedc551e378bdc"]/div/div[3]/p').text
merchant_2_name = browser.find_element_by_xpath('//*[@id="block-05ea3afedc551e378bdc"]/div/div[4]/span/b').text
merchant_2_wealth = browser.find_element_by_xpath('//*[@id="block-05ea3afedc551e378bdc"]/div/div[4]/p').text
richest_merchant_answer = browser.find_element_by_id('r3Input')
answer_button_3 = browser.find_element_by_id('r3Butn')

richest_merchant = ""
if (int(merchant_1_wealth) > int(merchant_2_wealth)):
    richest_merchant = str(merchant_1_name)
elif (int(merchant_1_wealth) < int(merchant_2_wealth)):
    richest_merchant = str(merchant_2_name)
else:
    richest_merchant = "neither"

richest_merchant_answer.send_keys(richest_merchant)
answer_button_3.click()

check_answers_button = browser.find_element_by_id('checkButn')
check_answers_button.click()
final_results = browser.find_element_by_xpath('//*[@id="successBanner2"]/h4').text

if (final_results == 'Success!'):
    print("\nV\n")
else:
    print('\nX\n')

# print("the final text was: " + str(final_results))

File: C#/TheRiddlePractice.cs
using System;
using OpenQA.Selenium;
using OpenQA.Selenium.Chrome;
using System.Threading;
class SauceLabsTest
{
    static void Main()
    {
        IWebDriver browser = new ChromeDriver();

        browser.Navigate().GoToUrl("https://techstepacademy.com/trial-of-the-stones");

        IWebElement riddleOfStoneTextField = browser.FindElement(By.Id("r1Input"));
        IWebElement answerButton1 = browser.FindElement(By.Id("r1Btn"));
        
        riddleOfStoneTextField.SendKeys("rock");
        answerButton1.Click();
        IWebElement answer1 = browser.FindElement(By.XPath(@"//*[@id='passwordBanner']/h4"));
        string answer1String = answer1.Text;

        IWebElement riddleOfSecretsTextField = browser.FindElement(By.Id("r2Input"));
        IWebElement answerButton2 = browser.FindElement(By.Id("r2Butn"));
        riddleOfSecretsTextField.SendKeys(answer1String);
        answerButton2.Click();
        IWebElement answer2 = browser.FindElement(By.XPath(@"//*[@id='successBanner1']/h4"));
        string answer2String = answer2.Text;

        IWebElement merchant1Name = browser.FindElement(By.XPath(@"//*[@id='block-05ea3afedc551e378bdc']/div/div[3]/span/b"));
        IWebElement merchant1Wealth = browser.FindElement(By.XPath("//*[@id='block-05ea3afedc551e378bdc']/div/div[3]/p"));
        IWebElement merchant2Name = browser.FindElement(By.XPath(@"//*[@id='block-05ea3afedc551e378bdc']/div/div[4]/span/b"));
        IWebElement merchant2Wealth = browser.FindElement(By.XPath(@"//*[@id='block-05ea3afedc551e378bdc']/div/div[4]/p"));
        IWebElement richestMerchantAnswer = browser.FindElement(By.Id("r3Input"));
        IWebElement answerButton3 = browser.FindElement(By.Id("r3Butn"));

        string merchant1NameString = merchant1Name.Text;
        int merchant1WealthInt = int.Parse(merchant1Wealth.Text);
        string merchant2NameString = merchant2Name.Text;
        int merchant2WealthInt = int.Parse(merchant2Wealth.Text);

        string richestMerchant = "";
        if (merchant1WealthInt > merchant2WealthInt)
        {
            richestMerchant = merchant1NameString;
        }
        else if (merchant1WealthInt < merchant2WealthInt)
        {
            richestMerchant = merchant2NameString;
        }
        else
        {
            richestMerchant = "none";
        }
        richestMerchantAnswer.SendKeys(richestMerchant);
        answerButton3.Click();

        IWebElement checkAnswersButton = browser.FindElement(By.Id("checkButn"));
        checkAnswersButton.Click();
        IWebElement finalResults = browser.FindElement(By.XPath(@"//*[@id='successBanner2']/h4"));
        string finalResultsString = finalResults.Text;

        if (finalResultsString.Equals("Success!"))
        {
            Console.WriteLine("\nV\n");
        }
        else
        {
            Console.WriteLine("\nX\n");
        }
        Console.WriteLine("The final text was: " + finalResultsString);
    }
}